ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -234,7 +234,7 @@ func (r *RayClusterReconciler) rayClusterReconcile(request ctrl.Request, instanc
 		}
 	}
 
-	return ctrl.Result{}, nil
+	return ctrl.Result{RequeueAfter: DefaultRequeueDuration}, nil
```

Comment:
What will happen in the following example?

(1) **t = 0**: `ctrl.Result{RequeueAfter: DefaultRequeueDuration}`
(2) **t = DefaultRequeueDuration / 2**: Delete RayCluster CR.
(3) **t = DefaultRequeueDuration**: Will the Pods be created again?

Reply:
@kevin85421 what's the desired behavior here? Is it something that should be addressed by this PR?

Reply:
The expected behavior is all related Kubernetes resources (e.g. Ray head Pod, Ray worker Pods, Ray head service, ... etc) should be removed and not be created again.

Reply:
> (2) **t = DefaultRequeueDuration / 2**: Delete RayCluster CR.
> (3) **t = DefaultRequeueDuration**: Will the Pods be created again?

I doubt this will happen (if it did we would have seen it by now).
I imagine we have a clause somewhere that skips reconciliation if the CR is marked for deletion?

Reply:
We should probably take a closer look at the internal KubeBuilder and controller-runtime.

Reply:
So in the controller, we need to first check if the CR exists, if (2) deletes the CR, then (3) knows the CR is deleted and then does not create pods again.

Reply:
I'd be surprised if this check is not already there somewhere.
We also need to check if `metadata.deletionTimestamp` is set on the CR and skip reconcile if it is (but again I'd be surprised if we don't do that already)

Reply:
I found a check here that skips reconcile if the deletion timestamp is set: https://github.com/ray-project/kuberay/blob/master/ray-operator/controllers/ray/raycluster_controller.go#L175-L178

I'm not sure where to look for the other check (the check if the CR exists), but I will manually test the scenario where the RayCluster CR is deleted.

Reply:
Verified manually by waiting for a RayJob to finish, then running `kubectl delete raycluster rayjob-sample-raycluster-d7k5v` and watching `kubectl get pods`. The pods get removed and no new pods get started, which is what we wanted.

Reply:
Suuper, thanks!